libobs/obs-hotkey.c

```diff
@@ -1233,12 +1233,18 @@ static inline void release_pressed_binding(obs_hotkey_binding_t *binding)
 	else if (obs->hotkeys.router_func)
 		obs->hotkeys.router_func(obs->hotkeys.router_func_data,
 					 hotkey->id, false);
+
+	if (hotkey->released_func)
+		hotkey->released_func(hotkey->data, hotkey->id, hotkey);
 }
 
 static inline void handle_binding(obs_hotkey_binding_t *binding,
 				  uint32_t modifiers, bool no_press,
 				  bool strict_modifiers, bool *pressed)
 {
+	if (pressed && !*pressed)
+		return;
```

Comment:
Not sure I understand what this is trying to fix since https://github.com/obsproject/obs-studio/blob/47b699e176132dbc51c03ad5a9a97a78b74ca198/libobs/obs-hotkey.c#L1264-L1266 already exists; did you run into a bad interaction with https://github.com/obsproject/obs-studio/blob/47b699e176132dbc51c03ad5a9a97a78b74ca198/libobs/obs-hotkey.c#L1255-L1256?

Reply:
When I comment out the modifiers_only lines, it still doesn't work. It only works when at the top of the function.

Reply:
I just moved the if (pressed && !*pressed) to the top of the function.

Reply:
Could you spell out what exactly "it" is, what setup are you using to trigger the behavior you're seeing?

Reply:
My use case is for holding the hotkey for a few seconds.

What I'm using it for specifically:
I've created a script to control a PTZ camera with hotkeys.

For example: 
When I press the hotkey, the camera will start panning. I would hold it down until I want to stop panning, and then when released the script would send a command to the camera to stop panning.